src/main/java/net/simon987/mar/cubot/CubotInventory.java

```diff
@@ -55,14 +55,19 @@ public void putItem(Item item) {
     private void scanItem() {
         int x = getCpu().getRegisterSet().getRegister("X").getValue();
         Item item = inventory.get(position);
-        item.digitize(unit.getCpu().getMemory(), x);
+        if (item != null) {
+            item.clear(unit);
```

Comment:
is this change intended? this looks like 2 added lines that are copy pasted from the changes below

Reply:
Nope, it was not. Nice catch!